src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -329,6 +360,18 @@ func (r *StatefulSetResource) Kind() string {
 	return statefulSetKind()
 }
 
+func (r *StatefulSetResource) portsConfiguration() string {
+	kafkaAPIPort := strconv.Itoa(r.pandaCluster.Spec.Configuration.KafkaAPI.Port)
+	rpcAPIPort := strconv.Itoa(r.pandaCluster.Spec.Configuration.RPCServer.Port)
+
+	// In every dns name there is trailing dot to query absolute path
+	// For trailing dot explanation please visit http://www.dns-sd.org/trailingdotsindomainnames.html
+	return "--advertise-kafka-addr=internal://$(POD_NAME)." + r.svc.Key().Name + ".$(POD_NAMESPACE).svc.cluster.local.:" + kafkaAPIPort +
```

Comment:
nit: Overall I prefer using fmt.Sprintf I feel like it's easier to see the result.

Reply:
I have no strong preference here :) I will change.